runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/control/FnApiControlClientPoolServiceTest.java

```diff
@@ -76,7 +76,9 @@ public void testIncomingConnection() throws Exception {
     StreamObserver<BeamFnApi.InstructionResponse> responseObserver =
         controlService.control(requestObserver);
 
-    FnApiControlClient client = pool.getSource().get();
+    // HACK: The Java Harness does not populate its own control client headers currently. They end
+    // up getting set to null by default, so we fetch clients by the null id.
```

Comment:
<!--new_thread; commit:1fd121da1417624b3b84f0300648251da64b9cb5; resolved:0-->
Ditto, reference the JIRA here too if any